src/NuGet.Clients/NuGet.PackageManagement.UI/Models/PackageSolutionDetailControlModel.cs

```diff
@@ -125,11 +125,27 @@ private async Task UpdateInstalledVersionsAsync(CancellationToken cancellationTo
         {
             var hash = new HashSet<NuGetVersion>();
 
-            foreach (var project in _projects)
+            IEnumerable<IProjectContextInfo> nugetProjects = _projects.Select(p => p.NuGetProject);
+            IReadOnlyDictionary<string, IReadOnlyCollection<IPackageReferenceContextInfo>> projectIdsToInstalledPackages =
+                await nugetProjects.GetInstalledPackagesAsync(ServiceBroker, cancellationToken);
+
+            foreach (KeyValuePair<string, IReadOnlyCollection<IPackageReferenceContextInfo>> item in projectIdsToInstalledPackages)
             {
+                string packageId = Id;
+                string projectId = item.Key;
+                IReadOnlyCollection<IPackageReferenceContextInfo> packageReferences = item.Value;
+                PackageInstallationInfo project = _projects.FirstOrDefault(installationInfo => StringComparer.OrdinalIgnoreCase.Equals(installationInfo.NuGetProject.ProjectId, projectId));
```

Comment:
I see we have a `PackageIdentityComparer`, but I'm surprised we don't have a `PackageIdComparer`.  It would simplify a lot of code if we could just use `PackageIdComparer.Instance`.

Reply:
I think you meant we should have a Project ID comparer, right? If so, I can create one. 
My quick search of the codebase did not find any other comparisons of project ID, though.

If you mean Package ID, then I think there's far too many references (over 150), and that sort of refactoring warrants a separate Issue and PR, which I'm happy to create.